### PR TITLE
Jenkins declarative syntax + timeout + disableConcurrentBuilds

### PR DESCRIPTION
**Pull request by 4ydan**
This should set a 2 hour timeout to the whole job.

## Basics

- [ ] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [ ] The PR is rebased with current master.
- [ ] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: ci/Jenkinsfile [hunk 1]
```diff
@@ -9,181 +9,183 @@ def wait_for_db() {
     }
 }
 
-// REQUIRED for Deployment
-// see Jenkinsfile.release
-stage('Build Schema'){
-    node('docker') {
-        node_info()
+timeout(time: 1, unit: 'MINUTES') {
+    disableConcurrentBuilds()
```
Comment (markus2330) [inline ci/Jenkinsfile:13]: Otherwise it would simply wait?

```suggestion
    disableConcurrentBuilds(abortPrevious: true)
```

**Comment by 4ydan** [comment]
I set the timeout to 1minute for this one commit, it seemed to work [here](https://build.libelektra.org/blue/organizations/jenkins/PermaPlanT-Folder%2FPermaplanT/detail/PR-553/5/pipeline). Will set it back to 2 hours.

**Review by markus2330 — COMMENTED** [review]
Thx for working on this!

File: ci/Jenkinsfile (hunk 1 above)
Comment (4ydan) [reply ci/Jenkinsfile]: Yea I was not aware how to do this, had to check the jenkins doc.

Comment (4ydan) [reply ci/Jenkinsfile]: Doesn't seem to work eitherway :/

**Comment by 4ydan** [comment]
So I thought I'll just switch to the declarative syntax and add the timeout and disableConcurrent thing there, since it will just look better when its outside or at the top of the pipeline, but now I see that we need to have docker installed on the permaplant nodes if we want to use a declarative syntax. Or maybe I m not using it right? I have the feeling that the nodes are not made so easily to swap to a declarative syntax.

image

image

**Comment by markus2330** [comment]
Thanks for looking into. I agree, it still seems to be only suitable for minimal CI setups. In Elektra we eventually also gave up using the declarative syntax. I hoped this would have changed, so I didn't immediately say that you shouldn't try to do it.

**Comment by 4ydan** [comment]
I found out why the declarative syntax was not working.
There is weird configuration inside the pipeline.
image
This should be docker in order for it to work. In opensesame we have a declarative pipeline now.

**Pull request closed without merging**